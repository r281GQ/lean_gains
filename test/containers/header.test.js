import { shallow } from 'enzyme';
import React from 'react';
import Header from './../../src/containers/header';

describe('Header container', () => {
  it('should render', () => {
    let name = 'tag';

    const mockClickHandler = jest.fn(bf => name = 'another');

    const wrapper = shallow(<Header label={name} clickHandler={mockClickHandler}/>)
    expect(wrapper.instance().props.label).toEqual('tag');

    // expect(wrapper).toMatchSnapshot();
    wrapper.find('button').simulate('click');
    const changed = shallow(<Header label={name} clickHandler={mockClickHandler}/>)
    expect(mockClickHandler).toBeCalled();
    expect(changed.instance().props.label).toEqual('another');
  });
});
