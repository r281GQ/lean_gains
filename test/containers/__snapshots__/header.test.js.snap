// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header container should render 1`] = `
<div>
  Header
</div>
`;
